app/Http/Controllers/ProfileController.php

```diff
@@ -132,4 +136,53 @@ public function destroy($id)
         return response()->json('success');
 
     }
+    
+    //Password change
+    public function changePassword(Request $request, $password)
+    {
+        //Authenticate user
+        $profile = Auth::user();
+
+        //Validation for password change
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'oldPassword' => 'required|min:8',
+                'newPassword' => 'required|min:8',
+                'repeatNewPassword' => 'required|min:8'
+
+            ]
+        );
+
+        if($validator->fails())
+        {
+            return response()->json(['error' => $validator->errors()->all(), 400]);
+        }
+
+        //Input old password, new password and repeat new password
+        $profile->oldPassword = $request->input('oldPassword');
+        $profile->newPassword = $request->input('newPassword');
+        $profile->repeatNewPassword = $request->input('repeatNewPassword');
+
+        if($newPassword != $repeatNewPassword)
+        {
+            return response()->json(['not the same password']);
+        }
+
+        if(Hash::make($password) !== $profile->password)
+        {
+            return response()->json('wrong password');
+        }
+
+        $this->newPassword = Hash::make($password);
```

Comment:
`$profile->password` is the way to set the `password` property onto loaded Profile. `$this` references the current class object, which is the ProfileController, and even if that where not the case, if it where the Profile object, `newPassword` is not a defined Profile field.